src/plugins/passwd/CMakeLists.txt

```diff
@@ -1,20 +1,20 @@
+include (CheckFunctionExists)
 include (LibAddMacros)
 
+set (plugin passwd)
+
 if (DEPENDENCY_PHASE)
-	try_compile (TEST_FGETPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasfgetpwent.c")
-	try_compile (TEST_PUTPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasputpwent.c")
+	check_function_exists (fgetpwent TEST_FGETPWENT)
+	check_function_exists (putpwent TEST_PUTPWENT)
+	check_function_exists (getpwent TEST_GETPWENT)
 
-	if (TEST_PUTPWENT)
-		set (PUTPWENT 1)
-	else ()
-		set (PUTPWENT 0)
-	endif ()
 	if (TEST_FGETPWENT)
-		set (FGETPWENT 1)
+		add_definitions (-D_GNU_SOURCE)
+	elseif (TEST_PUTPWENT AND TEST_GETPWENT)
+		add_definitions (-D_X_OPEN_SOURCE=500)
 	else ()
-		set (FGETPWENT 0)
+		remove_plugin (${plugin} "Could not find required functions")
```

Comment:
error message unclear: Which functions are this? Where they are provided? (I assume glibc?)

Reply:
`check_function_exists` already logs output hence I did not specify the functions here again:
```
-- Looking for fgetpwent
-- Looking for fgetpwent - found
-- Looking for putpwent
-- Looking for putpwent - found
-- Looking for getpwent
-- Looking for getpwent - found
-- Include Plugin passwd
```
And no not glibc: it just needs to be available in the include paths (typically it is provided by your libc implementation, if you are use something at least somewhat POSIX).

Reply:
You only see this output in the first cmake run. In the later runs you only get the error. So the error message needs to describe what the reason is.